### ISSUE TITLE
E1012 incorrectly raised when using Fn::ForEach with !FindInMap

### ISSUE THREAD

**Issue by verbitan** [aws-cloudformation/cfn-lint#2840]
### CloudFormation Lint Version

0.79.6

### What operating system are you using?

Amazon Linux 2023

### Describe the bug

Using `!FindInMap` within `Fn::ForEach` is throwing the error `E1012: Ref Microservice not found as a resource or parameter`. This doesn't occur on all `!FindInMap` instances.

In the example template, this occurs on the following line.
```yaml
          Conditions:
            - Field: path-pattern
              PathPatternConfig:
                Values: !FindInMap
                  - MicroserviceConfig
                  - !Ref Microservice
                  - FilterPatternConfig
```

But not these lines.
```yaml
          Port: !FindInMap
            - MicroserviceConfig
            - !Ref Microservice
            - Port
...
          HealthCheckPath: !FindInMap
            - MicroserviceConfig
            - !Ref Microservice
            - HealthCheckPath
...
          Priority: !FindInMap
            - MicroserviceConfig
            - !Ref Microservice
            - Priority
```

### Expected behavior

The template deploys as expected via CloudFormation, so I expect `cfn-lint` to be ok with this template also.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Transform: AWS::LanguageExtensions

Parameters:
  VPCID:
    Type: AWS::EC2::VPC::Id

  ApplicationLoadBalancer:
    Type: String

  ApplicationLoadBalancerListener:
    Type: String

  ApplicationLoadBalancerCertificate:
    Type: String

Mappings:
  MicroserviceConfig:
    ExmapleService:
      Priority: 1
      Port: 9000
      FilterPatternConfig:
        - /some-path*
        - /some-other-path*
      HealthCheckPath: /some-path/healthcheck

Resources:
  Fn::ForEach::Microservices:
    - Microservice
    - - ExampleService
    - ${Microservice}LoadBalancerListener:
        Type: AWS::ElasticLoadBalancingV2::Listener
        Properties:
          LoadBalancerArn: !Ref ApplicationLoadBalancer
          Protocol: HTTPS
          Port: !FindInMap
            - MicroserviceConfig
            - !Ref Microservice
            - Port
          SslPolicy: ELBSecurityPolicy-FS-1-2-Res-2020-10
          Certificates:
            - CertificateArn: !Ref ApplicationLoadBalancerCertificate
          DefaultActions:
            - Type: forward
              TargetGroupArn: !Ref
                Fn::Sub: ${Microservice}LoadBalancerTarget

      ${Microservice}LoadBalancerTarget:
        Type: AWS::ElasticLoadBalancingV2::TargetGroup
        Properties:
          VpcId: !Ref VPCID
          Protocol: HTTPS
          Port: 8443
          HealthCheckPath: !FindInMap
            - MicroserviceConfig
            - !Ref Microservice
            - HealthCheckPath
          TargetType: instance

      ${Microservice}ListenerRule:
        Type: AWS::ElasticLoadBalancingV2::ListenerRule
        Properties:
          Actions:
            - Type: forward
              ForwardConfig:
                TargetGroups:
                  - TargetGroupArn: !Ref
                      Fn::Sub: ${Microservice}LoadBalancerTarget
                    Weight: 1
          Conditions:
            - Field: path-pattern
              PathPatternConfig:
                Values: !FindInMap
                  - MicroserviceConfig
                  - !Ref Microservice
                  - FilterPatternConfig
          ListenerArn: !Ref ApplicationLoadBalancerListener
          Priority: !FindInMap
            - MicroserviceConfig
            - !Ref Microservice
            - Priority
```
